src/components/darkmode-switch/darkmode-switch.tsx

```diff
@@ -0,0 +1,66 @@
+import { Component, h, State } from '@stencil/core';
+
+/**
+ * @private
+ */
+@Component({
+    tag: 'kompendium-darkmode-switch',
+    styleUrl: 'darkmode-switch.scss',
+    shadow: true,
+})
+export class DarkmodeSwitch {
+    @State()
+    private theme: string;
+
+    private select: HTMLSelectElement;
+
+    public componentWillLoad() {
+        const value = localStorage.getItem('theme') || 'system-default';
+        this.setTheme(value);
+    }
+
+    public componentDidLoad() {
+        switch (this.theme) {
+            case 'system-default':
+                this.select.selectedIndex = 0;
+                break;
+            case 'force-light':
+                this.select.selectedIndex = 1;
+                break;
+            case 'force-dark':
+                this.select.selectedIndex = 2;
+                break;
+            default:
+                this.select.selectedIndex = 0;
+        }
+    }
+
+    public render(): HTMLElement {
+        return (
+            <select ref={this.getSelectRef} onChange={this.handleThemeChange}>
```

Comment:
I love this but I'm not sure if we should do it with the `select`. I'd rather have a toggle button that is either light or dark (based on the system or browser setting. [which is Auto basically]), and user can toggle to the other mode if they want.

For inspiration, check this toggle button on the top right corner, for [another documentation tool](https://github.com/surveylegend/hapi-docs) that I designed: https://api.surveylegend.com/docs/#intro a few years ago, (for a startup which exploited and wasted precious 10 years of my life, which I deeply regret now).

There is also the matter of remembering user's choice, but that's another story. Maybe you have already addressed it.

Reply:
I'm happy to make it look better if you provide me with an icon or two 😊

Reply:
there are no icons there. I made the entire toggle with CSS ;) Just put a checbox or something and I'll style it even cooler ;)

Reply:
Ok, I'll fix 👍

Reply:
Now there's a checkbox instead. You check it to override the mode the system requests. Currently it gets a class `force-light` or `force-dark` when it's checked, and `system-default` when it's unchecked. Please tell me if you need something else or something more.

Reply:
And, yes, it does remember your setting.

Reply:
⚡

Reply:
@Kiarokh Can you approve this?